### PR TITLE
Handling missing main method on startup and `here.main` syntax

### PR DESCRIPTION
**Pull request by mwu-tow**
### Pull Request Description
Currently IDE fails to load if `main` method definition in not present or if it uses the `here.main` syntax.
This PR fixes both:
1) method pointer lookup will correctly take into consideration the module name to resolve the `here` symbol (related engine issue: https://github.com/enso-org/enso/issues/1138 ) 
2) if the `main` method cannot be found during the startup, it will be created

Currently I am marking this PR as a draft, because there's one more missing piece — the standard `main` definition stub requires also explicit `Base` import.
To be discussed how to handle this.

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Rust](https://github.com/luna/enso/blob/main/docs/style-guide/rust.md).
- [x] All code has been tested where possible.
- [ ] All code has been profiled where possible.

### CODE REVIEW

File: src/rust/ide/src/double_representation/module.rs [hunk 1]
```diff
@@ -598,18 +608,27 @@ pub fn locate
...
 /// desugaring implicit extensions methods for modules).
+///
+/// The `name` parameter is the name of the module that contains `ast`. It affects how the `here`
+/// keyword is resolved.
 pub fn lookup_method
-(ast:&known::Module, method:&language_server::MethodPointer) -> FallibleResult<definition::Id> {
-    let module_name                    = QualifiedName::try_from(method)?;
-    let explicitly_extends_looked_type = method.defined_on_type == module_name.name().as_ref();
-
+(name:&ReferentName, ast:&known::Module, method:&language_server::MethodPointer)
```
Comment (farmaazon) [inline src/rust/ide/src/double_representation/module.rs:615]: I would rename `name` parameter to `module_name`

File: src/rust/ide/src/double_representation/module.rs [hunk 2]
```diff
@@ -598,18 +608,27 @@ pub fn locate
 ///
 /// The module is assumed to be in the file identified by the `method.file` (for the purpose of
 /// desugaring implicit extensions methods for modules).
+///
+/// The `name` parameter is the name of the module that contains `ast`. It affects how the `here`
+/// keyword is resolved.
 pub fn lookup_method
```
Comment (farmaazon) [inline src/rust/ide/src/double_representation/module.rs:614]: At this point when I read this module I wonder what decides when a function is a method of `Info` and when it isn't? Perhaps most of them could go inside Info, and even Info could easily keep module name next to ast?

That only under your consideration, it is not blocker for merging PR.

File: src/rust/ide/src/view/project.rs [hunk 3]
```diff
@@ -91,20 +110,44 @@ pub async fn recreate_if_missing(project:&model::Project, path:&FilePath, defaul
...
+pub fn add_main_if_missing
+(module:&model::Module, main_ptr:&MethodPointer, parser:&Parser) -> FallibleResult<()> {
+    if module.lookup_method(main_ptr).is_err() {
+        let mut info  = module.info();
+        let main_code = default_main_method_code();
+        let main_ast  = parser.parse_line(main_code)?;
+        info.add_ast(main_ast,double_representation::module::Placement::End)?;
+        module.update_ast(info.ast);
+
+        // TODO [mwu] Add imports required by the default main definition (when needed).
```
Comment (farmaazon) [inline src/rust/ide/src/view/project.rs:125]: Let's add it right now, or put here a reference to issue with managing new import syntax.

File: src/rust/ide/src/view/project.rs [hunk 4]
```diff
@@ -91,20 +110,44 @@ pub async fn recreate_if_missing(project:&model::Project, path:&FilePath, defaul
...
+    }
+    Ok(())
+}
+
 impl ProjectView {
     /// Create a new ProjectView.
-    pub async fn new(logger:impl AnyLogger, model:model::Project)
-    -> FallibleResult<Self> {
+    pub async fn new(logger:impl AnyLogger, project:model::Project)
+                     -> FallibleResult<Self> {
```
Comment (farmaazon) [inline src/rust/ide/src/view/project.rs:133]:
```suggestion
    -> FallibleResult<Self> {
```

**Review by farmaazon — APPROVED** [review]

File: src/rust/ide/src/double_representation/module.rs (hunk 2 above)
Comment (mwu-tow) [reply src/rust/ide/src/double_representation/module.rs:614]: Leaving as-is, should be refactored in the future.

Functions being free and not in `Info` is mostly consequence of `Info` being introduced later than them. We don't want `Info` to contain name, as most of the methods don't need it.  
Another issue is the ergonomy of usage — having to wrap Ast into info every single time is not convenient. Some places also don't know name but want to process Ast.

The lookup function however should not really be a method should not be a method of plain info but something that contains both ast and name.  If other functions working on such name+ast pair appear, we should consider introducing it as a new structure (NamedInfo or sth similar). Info should not be structurally changed.

File: src/rust/ide/src/double_representation/module.rs (hunk 1 above)
Comment (mwu-tow) [reply src/rust/ide/src/double_representation/module.rs:615]: done

File: src/rust/ide/src/view/project.rs (hunk 4 above)
Comment (mwu-tow) [reply src/rust/ide/src/view/project.rs:133]: actually this can fit in a single line

**Comment by mwu-tow** [comment]
Had a call about imports issue. Decided to change the contents of the default generated main to just contain `"Hello world!" node with no dependency on the standard library, so the import won't be needed.

**Pull request merged**